Three-D-Viewer-Server/src/main/java/com/main/server/security/JwtTokenProvider.java

```diff
@@ -0,0 +1,88 @@
+package com.main.server.security;
+
+import com.main.server.entity.Role;
+import com.main.server.exception.JwtAuthenticationException;
+import io.jsonwebtoken.*;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+
+@Component
+@AllArgsConstructor
+public class JwtTokenProvider {
+
+    @Value("${jwt.token.secret}")
+    private String secretKey;
+    @Value("${jwt.token.expired}")
+    private Long validityInMilliseconds;
+    private final UserDetailsService userDetailsService;
+
+    @PostConstruct
+    protected void init() {
+        secretKey = Base64.getEncoder().encodeToString(secretKey.getBytes());
+    }
+
+    public String createToken(String username, List<Role> role) {
```

Comment:
Ok, it is good. But you also need the add generation of refreshToken, The token can be in the format of jwt token or you can find other implementation